app/services/data_migrations/backfill_application_choices_with_work_experiences.rb

```diff
@@ -0,0 +1,34 @@
+module DataMigrations
+  class BackfillApplicationChoicesWithWorkExperiences
+    TIMESTAMP = 20240822105049
+    MANUAL_RUN = true
+
+    def change
+      time_now = Time.zone.now
+
+      choices_without_work_histories.each_slice(5000).with_index do |ids, index|
+        next_batch_time = time_now + index.minutes
```

Comment:
<details>
<summary>I think this will cause the last job to be scheduled nearly 2hours from now, is this what we want to happen?</summary>

592,000 records, split by 5000 (batch size) = 118 max index value

</details>

Reply:
I'm not sure. Initially I thought we will enqueue a lot of jobs and the sidekiq workers will just go through the enqueued jobs and finish whenever they finish.

But I see that similar migrations have been using [BatchDelivery](https://github.com/DFE-Digital/apply-for-teacher-training/blob/main/app/services/batch_delivery.rb) class which spreads jobs over a period of time. I can't use that because of the different objects coming back from `find_by_sql` so I thought I would try to replicate the behaviour of the `BatchDelivery` class.

Personally I think the workers will take longer than 2 hours to migrate all the data just because they have to go though each record and use ActiveRecord.

I'm happy to reduce this, leave it or just remove it completely.